src/AnalyticsEventReporter.ts

```diff
@@ -66,6 +83,20 @@ export class AnalyticsEventReporter implements AnalyticsEventService {
     const shouldUseBeacon = useBeacon(finalPayload, this.config.forceFetch);
     const requestUrl = setupRequestUrl(this.config.env, this.config.region);
 
+    if (this.config.debug) {
+      console.log(
+        '[DEBUG] The following Payload would be sent to the Yext Events API using ' +
+          (shouldUseBeacon ? 'Beacon. ' : 'fetch(). ') +
+          'Session Tracking is ' +
+          (this.config.sessionTrackingEnabled ? 'enabled.\n' : 'disabled.\n') +
+          'Request URL: ' +
+          requestUrl +
+          '\n' +
+          'To send the event to the Yext Events API, disable the debug flag in your AnalyticsConfig.\n',
+        finalPayload
+      );
+    }
+
     // If useBeacon returns true, return boolean response of postWithBeacon as string.
     if (shouldUseBeacon) {
       return new Promise((resolve, reject) => {
```

Comment:
I know we discussed to do this in two steps: add logging to the sdk and drop debug mode requests in PixelGateway, but I think we should still not send any requests if it's in debug mode.

Reply:
Yeah that's my bad I meant to put the post calls behind the debug flag.